controllers/api/pwa.js

```diff
@@ -128,34 +160,48 @@ const rssWriter = new RssWriter();
  *
  * Returns all PWAs as JSON or ?format=csv for CSV.
  */
-router.get('/', checkApiKey, (req, res) => {
+router.get('/:id*?', (req, res) => {
   let format = req.query.format || 'json';
   let sort = req.query.sort || 'newest';
   let skip = parseInt(req.query.skip, 10);
-  let limit = parseInt(req.query.limit, 10);
-
+  let limit = parseInt(req.query.limit, 10) || 100;
   res.setHeader('Cache-Control', 'public, max-age=' + CACHE_CONTROL_EXPIRES);
-  pwaLib.list(skip, limit, sort)
-    .then(result => {
-      switch (format) {
-        case 'csv': {
-          csvWriter.write(res, result.pwas);
-          break;
-        }
-        case 'rss': {
-          rssWriter.write(res, result.pwas);
-          break;
-        }
-        default: {
-          jsonWriter.write(res, result.pwas);
-        }
+
+  return new Promise((resolve, reject) => {
```

Comment:
We could simplify this with something like:

```js
let promise;
if (req.params.id) {
  promise = findPwa(req.params.id);
} else {
  promise = pwaLib.list(...)
}
promise
  .then(result) => {
    ...
  }

```

Reply:
Good catch! fixed.